include/gridtools/storage/adapter/nanobind_adapter.hpp

```diff
@@ -24,76 +24,86 @@
 
 namespace gridtools {
     namespace nanobind_sid_adapter_impl_ {
+#if NB_VERSION_MAJOR >= 2
+    using _ssize_t = nanobind::ssize_t;
+#else
+    using _ssize_t = std::size_t;
+#endif
 
-        // Use `-1` for dynamic stride, use an integral value for static stride.
-        template <nanobind::ssize_t... Values>
-        using stride_spec = std::integer_sequence<nanobind::ssize_t, Values...>;
-
-        template <class IndexSequence>
-        struct dynamic_strides_helper;
-
-        template <std::size_t... Indices>
-        struct dynamic_strides_helper<std::index_sequence<Indices...>> {
-            using type = stride_spec<(void(Indices), -1)...>;
-        };
-
-        template <std::size_t N>
-        using fully_dynamic_strides = typename dynamic_strides_helper<std::make_index_sequence<N>>::type;
-
-        template <nanobind::ssize_t SpecValue>
-        auto select_static_stride_value(std::size_t dyn_value) {
-            if constexpr (SpecValue == -1) {
-                return dyn_value;
-            } else {
-                if (SpecValue != dyn_value) {
-                    throw std::invalid_argument("static stride in stride specification doesn't match dynamic stride");
-                }
-                return gridtools::integral_constant<std::size_t, SpecValue>{};
+    inline constexpr nanobind_sid_adapter_impl_::_ssize_t dynamic_size =
+        static_cast<nanobind_sid_adapter_impl_::_ssize_t>(-1);
```

Comment:
Isn’t the following good enough (and much more readable)?
```suggestion
    inline constexpr _ssize_t dynamic_size = -1;
```

Reply:
that was a relict of moving things around